src/extension/appLauncher.ts

```diff
@@ -225,10 +225,17 @@ export class AppLauncher {
                                 return mobilePlatform.runApp();
                             })
                             .then(() => {
-                                if (mobilePlatformOptions.isDirect) {
-                                    generator.step("mobilePlatform.enableDirectDebuggingMode");
-                                    if (launchArgs.platform === "android") {
-                                        this.logger.info(localize("PrepareHermesDebugging", "Prepare Hermes debugging (experimental)"));
+                                if (mobilePlatformOptions.isDirect || !mobilePlatformOptions.debuggingEnabled) {
```

Comment:
Do we need that `!mobilePlatformOptions.debuggingEnabled` check? We also have [this](https://github.com/microsoft/vscode-react-native/pull/1319/files#diff-2c75a606ef72d09ef7d108ed6b807b7cR231) after that check and that confused me a bit.